ftdc/custom_format_test.go

```diff
@@ -152,8 +153,116 @@ func TestCustomFormatRoundtripRich(t *testing.T) {
 }
 
 func TestReflection(t *testing.T) {
-	test.That(t, getFieldsForItem(&Basic{100}), test.ShouldResemble,
+	fields, err := getFieldsForStruct(reflect.TypeOf(&Basic{100}))
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, fields, test.ShouldResemble,
 		[]string{"Foo"})
-	test.That(t, getFieldsForItem(&Statser1{100, 0, 44.4}), test.ShouldResemble,
+
+	fields, err = getFieldsForStruct(reflect.TypeOf(Statser1{100, 0, 44.4}))
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, fields, test.ShouldResemble,
 		[]string{"Metric1", "Metric2", "Metric3"})
 }
+
+type TopLevel struct {
```

Comment:
Also curious what embedded/promoted fields look like in the FTDC format. I guess all depends on what `structField.Name` returns for those; I'm assuming it's just the name of the type.

Reply:
Is [this test](https://github.com/viamrobotics/rdk/pull/4465/files#diff-132974664df279d8a53ead66fc581c81749f662ba7f64c6e1d06b85fb99adf62R184) case answering your question?

Reply:
I guess not exactly; neither `TopLevel` nor `Nested` have embedded fields (maybe I'm using the wrong term). I meant something like
```go
type Embedded struct {
     Nested // embedded, unnamed field
     X int
}
```
Was curious what the string for `Nested` would be in that case (I'm assuming just `"Nested"`).